### PR TITLE
Multiple sign in refactor

### PR DESCRIPTION
**Pull request by karthikmsqa**
# Description

This PR includes the following proposed change(s):

- Refactored multiple sign in related classes

## Type of change

- [x] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

- Local host on docker demo

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: tests/src/test/java/ca/bc/gov/open/jag/efiling/TestConfig.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public AuthenticationPage bcscAuthenticationPageImpl() {
     }
 
     public AuthenticationPage keycloakAuthenticationPageImpl() {
-        return new KeycloakAuthenticationPageImpl(generateUrlService());
+        return new KeycloakAuthenticationPageImpl();
```
Comment (alexjoybc) [inline tests/src/test/java/ca/bc/gov/open/jag/efiling/TestConfig.java:111]: Nice!

**Review by TayGov — APPROVED** [review]

File: tests/src/test/java/ca/bc/gov/open/jag/efiling/stepDefinitions/AuthenticateAndRedirectToEfilingHubSD.java [hunk 2]
```diff
@@ -2,40 +2,57 @@
...
+        if (this.authenticationPage.getName().equalsIgnoreCase("keycloak")) {
+            if (this.generateUrlService.getGeneratedUrl() == null)
+                throw new EfilingTestException("Redirect url is not generated.");
+            this.efilingAdminHomePage.goTo(this.generateUrlService.getGeneratedUrl());
+        } else {
+            this.efilingAdminHomePage.goTo(efilingAdminUrl);
+        }
+        logger.info("Waiting for the page to load...");
+
+        this.authenticationPage.signIn();
```
Comment (alexjoybc) [inline tests/src/test/java/ca/bc/gov/open/jag/efiling/stepDefinitions/AuthenticateAndRedirectToEfilingHubSD.java:52]: I would pass the url in this method and and then call go to in the sign in method

**Review by alexjoybc — CHANGES_REQUESTED** [review]
looks @karthikmsqa a small comment. this looks really good

File: tests/src/test/java/ca/bc/gov/open/jag/efiling/stepDefinitions/AuthenticateAndRedirectToEfilingHubSD.java (hunk 2 above)
Comment (karthikmsqa) [reply tests/src/test/java/ca/bc/gov/open/jag/efiling/stepDefinitions/AuthenticateAndRedirectToEfilingHubSD.java:52]: Thanks @alexjoybc! will open a new PR for this.

**Pull request merged**